### ISSUE TITLE
Accessing Headers/Response on Execptions

### ISSUE THREAD

**Issue by BlakeLucchesi** [dropbox/dropbox-sdk-dotnet#38]
Is there a recommended method for accessing the response and/or response headers for calls that trigger exceptions?  I was looking through the DropboxRequestHandler.cs file and it doesn't look like any of the values from the HTTP response are passed along into the exceptions that get thrown (RetryException, BadInputException).  

It looks like the RateLimitException does a nice job of passing back along the Retry-After header as part of the structured exception.  I'm looking to read the "X-Dropbox-Request-Id " and "X-Server-Response-Time" headers in my exception handling code.

**Comment by greg-db** [comment]
I don't believe the SDK exposes the response headers like this, but we'll consider it a feature request.
